cmd/query/app/query_parser.go

```diff
@@ -46,22 +49,64 @@ const (
 var (
 	errMaxDurationGreaterThanMin = fmt.Errorf("'%s' should be greater than '%s'", maxDurationParam, minDurationParam)
 
-	// ErrServiceParameterRequired occurs when no service name is defined
-	ErrServiceParameterRequired = fmt.Errorf("parameter '%s' is required", serviceParam)
+	// errServiceParameterRequired occurs when no service name is defined.
+	errServiceParameterRequired = fmt.Errorf("parameter '%s' is required", serviceParam)
 )
 
-// queryParser handles the parsing of query parameters for traces
-type queryParser struct {
-	traceQueryLookbackDuration time.Duration
-	timeNow                    func() time.Time
-}
+type (
+	// queryParser handles the parsing of query parameters for traces.
+	queryParser struct {
+		traceQueryLookbackDuration time.Duration
+		timeNow                    func() time.Time
+	}
+
+	traceQueryParameters struct {
+		spanstore.TraceQueryParameters
+		traceIDs []model.TraceID
+	}
+
+	dependenciesQueryParameters struct {
+		endTs    time.Time
+		lookback time.Duration
+	}
+
+	durationParser interface {
```

Comment:
embrace functional programming - this is just a function, and the two parsers can be easily implemented to return functions, without using any structs